src/overlays/actors/ovl_Bg_Hidan_Sima/z_bg_hidan_sima.c

```diff
@@ -62,28 +68,243 @@ static ColliderJntSphInit D_8088F1E8 = {
+    } else {
+        this->actionFunc = func_8088E6D0;
+        this->dyna.actor.world.pos.x = this->dyna.actor.home.pos.x;
+        this->dyna.actor.world.pos.z = this->dyna.actor.home.pos.z;
+    }
+    if (!(this->timer % 4)) {
+        func_800AA000(this->dyna.actor.xyzDistToPlayerSq, 180, 10, 100);
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_BLOCK_SHAKE);
+    }
+}
+
+void func_8088E6D0(BgHidanSima* this, GlobalContext* globalCtx) {
+    if (func_8004356C(&this->dyna)) {
+        this->timer = 20;
+    } else if (this->timer != 0) {
+        this->timer--;
+    }
+    Math_StepToF(&this->dyna.actor.world.pos.y, this->dyna.actor.home.pos.y - 100.0f, 1.7f);
+    if (this->timer == 0) {
+        this->actionFunc = func_8088E518;
+    }
+}
+
+void func_8088E760(BgHidanSima* this, GlobalContext* globalCtx) {
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    if (this->timer == 0) {
+        this->dyna.actor.world.rot.y += 0x8000;
+        this->timer = 60;
+        this->actionFunc = func_8088E7A8;
+    }
+}
+
+void func_8088E7A8(BgHidanSima* this, GlobalContext* globalCtx) {
+    f32 temp;
+
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    if (this->dyna.actor.world.rot.y != this->dyna.actor.home.rot.y) {
+        temp = (sinf(((60 - this->timer) * 0.01667 - 0.5) * M_PI) + 1) * 200;
+    } else {
+        temp = (sinf((this->timer * 0.01667 - 0.5) * M_PI) + 1) * -200;
+    }
+    this->dyna.actor.world.pos.x = Math_SinS(this->dyna.actor.world.rot.y) * temp + this->dyna.actor.home.pos.x;
+    this->dyna.actor.world.pos.z = Math_CosS(this->dyna.actor.world.rot.y) * temp + this->dyna.actor.home.pos.z;
+    if (this->timer == 0) {
+        this->timer = 20;
+        this->actionFunc = func_8088E760;
+    }
+    func_8002F974(&this->dyna.actor, NA_SE_EV_FIRE_PILLAR - SFX_FLAG);
+}
+
+void func_8088E90C(BgHidanSima* this) {
+    s32 pad[2];
+    f32 cos;
+    f32 sin;
+    s32 i;
+
+    cos = Math_CosS(this->dyna.actor.world.rot.y + 0x8000);
+    sin = Math_SinS(this->dyna.actor.world.rot.y + 0x8000);
+
+    for (i = 0; i < 2; i++) {
+        ColliderJntSphElement* elem = &this->collider.elements[i];
+
+        elem->dim.worldSphere.center.x = this->dyna.actor.world.pos.x + sin * elem->dim.modelSphere.center.z;
+        elem->dim.worldSphere.center.y = (s16)this->dyna.actor.world.pos.y + elem->dim.modelSphere.center.y;
+        elem->dim.worldSphere.center.z = this->dyna.actor.world.pos.z + cos * elem->dim.modelSphere.center.z;
+    }
+}
+
+void BgHidanSima_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgHidanSima* this = THIS;
+    s32 pad;
+
+    this->actionFunc(this, globalCtx);
+    if (this->dyna.actor.params != 0) {
+        s32 temp;
+        if (this->dyna.actor.world.rot.y == this->dyna.actor.shape.rot.y) {
+            temp = this->timer;
+        } else {
+            temp = this->timer + 80;
+        }
```

Comment:
```suggestion
        s32 temp = (this->dyna.actor.world.rot.y == this->dyna.actor.shape.rot.y) ? this->timer : (this->timer + 80)
```
assuming it matches and it's a way we want to write it.